Doc/library/sqlite3.rst

```diff
@@ -1591,6 +1575,22 @@ Cursor objects
       including :abbr:`CTE (Common Table Expression)` queries.
       It is only updated by the :meth:`execute` and :meth:`executemany` methods.
 
+   .. attribute:: row_factory
+
+      Control how a row fetched from this :class:`!Cursor` is represented.
+      If ``None``, a row is represented as a :class:`tuple`.
+      Can be set to the included :class:`sqlite3.Row`;
+      or a :term:`callable` that accepts two arguments,
+      a :class:`Cursor` object and the :class:`!tuple` of row values,
+      and returns a custom object representing an SQLite row.
```

Comment:
I think this part could be removed, especially if `Connection.row_factory` is preferred.

IOW I would mimic the suggested phrasing of `Connection.row_factory` and say that:
* you can assign a callable
* the default value is `None`, which returns a `tuple`
* other cursors (for `Connection.row_factory`) and the parent connection (for `Cursor.row_factory`) are not affected
* there is an howto with more info

Then in the howto you can show how to assign an existing callable (e.g. `Row`) or how to define your own (and what args it should accept) and provide more info and examples.

Reply:
Yes, this was already discussed above. Here's my reasoning for the current layout:

As per Diataxis, the reference should be accurate and to the point; it should avoid conversations and digressions. It should describe, not explain. We should also avoid duplication, as far as possible.

`Connection.row_factory` has one job: it provides the initial `Cursor.row_factory` for cursors created from the connection.
The `Connection.row_factory` reference should explain this, and only this. It should also link to `Cursor.row_factory` and to the row factory how-to.

`Cursor.row_factory`, on the other hand, performs the work of adapting "row tuples" to custom objects, if set. If `None`, there is no row factory; rows are returned as tuples. If not `None`, it must be set to a callable with a specific argument spec. The `Cursor.row_factory` reference should explain all these things. It should also link to `Connection.row_factory` and to the row factory how-to.

I'm not sure it is a problem that we recommend setting the row factory on the connection object instead of on cursors. Also, I think there is a slight difference in the words "recommend" and "prefer". We recommend setting row factory on the connection, because it is practical for the user. We could rephrase that sentence, but I'm afraid it would quickly become too verbose.


> IOW I would mimic the suggested phrasing of `Connection.row_factory` and say that:
> * you can assign a callable

Yep, and we already say that :)

> * the default value is `None`, which returns a `tuple`

The initial / default value is whatever the parent Connection `row_factory` is.

> * other cursors (for `Connection.row_factory`) and the parent connection (for `Cursor.row_factory`) are not affected

We already say that the parent connection is not affected. I'm not sure it is worth it saying that other cursors are not affected. IMO that's implicit, since this is a normal attribute, not a class variable.

> * there is an howto with more info

We also say that ;)

Reply:
I believe we can resolve this conversation. Shout out if you disagree :)